acl/policy.go

```diff
@@ -352,6 +407,25 @@ func Parse(rules string) (*Policy, error) {
 
 	}
 
+	for _, np := range p.NodePools {
+		if !validNodePool.MatchString(np.Name) {
+			return nil, fmt.Errorf("Invalid node pool name: %#v", np)
```

Comment:
Won't this result in a go struct representation being written to logs? The structs aren't usually user-facing and aren't really 1:1 with the HCL. I think we probably want to report the specific name that's invalid here. Similar for the "invalid node pool policy" error below -- let's tell the user why it's invalid (in this case that it has an invalid capability)

Reply:
Ah that's right, it was a copy-pasta from other rule types. We're not very consistent how this is reported so I will create another format for now 😅